RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,278 @@
+                                           const SiStripNoises& noises,
+                                           std::vector<float>& invthick,
+                                           std::vector<uint32_t>& detID,
+                                           std::vector<uint16_t>& iPair,
+                                           std::vector<uint16_t>& noise,
+                                           std::vector<float>& gain);
+  };
+
+  void SiStripClusterizerConditionsESProducerAlpaka::fillSiStripClusterizerConditions_(const SiStripQuality& quality,
+                                                                                       const SiStripGain* gains,
+                                                                                       const SiStripNoises& noises,
+                                                                                       std::vector<float>& invthick,
+                                                                                       std::vector<uint32_t>& detID,
+                                                                                       std::vector<uint16_t>& iPair,
+                                                                                       std::vector<uint16_t>& noise,
+                                                                                       std::vector<float>& gain) {
+    // Lambda to lookup channel index and set the invThickness array
+    auto setInvThickness = [&](uint16_t fed, uint8_t channel, float val) {
+      invthick[channelIndex(fed, channel)] = val;
+    };
+
+    // connected: map<DetID, std::vector<int>>
+    // map of KEY=detid DATA=vector of apvs, maximum 6 APVs per detector module :
+    const auto& connected = quality.cabling()->connected();
+    // detCabling: map<DetID, std::vector<const FedChannelConnection*>
+    // map of KEY=detid DATA=vector<FedChannelConnection>
+    const auto& detCabling = quality.cabling()->getDetCabling();
+
+    for (const auto& conn : connected) {
+      const auto det = conn.first;
+      if (!quality.IsModuleBad(det)) {
+        const auto detConn_it = detCabling.find(det);
+
+        if (detCabling.end() != detConn_it) {
+          for (const auto& chan : (*detConn_it).second) {
+            if (chan && chan->fedId() && chan->isConnected()) {
+              const auto chan_detID = chan->detId();
+              const auto chan_fedID = chan->fedId();
+              const auto chan_fedCh = chan->fedCh();
+              const auto chan_apvPairNumber = chan->apvPairNumber();
+
+              // Fill the data structures
+              // Note: the channelIndex is used to access the data structures
+              detID[channelIndex(chan_fedID, chan_fedCh)] = chan_detID;
+              iPair[channelIndex(chan_fedID, chan_fedCh)] = chan_apvPairNumber;
+              setInvThickness(chan_fedID, chan_fedCh, siStripClusterTools::sensorThicknessInverse(chan_detID));
```

Comment:
why not
```suggestion
              invthick[channelIndex(chan_fedID, chan_fedCh)] = siStripClusterTools::sensorThicknessInverse(chan_detID);
```
like the two above ?